### PR TITLE
chore: upgrade detekt and ktlint

### PR DESCRIPTION
**Pull request by clemstoquart**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -22,9 +22,9 @@ repositories {
 }
 
 dependencies {
-  implementation("org.sonarsource.scanner.gradle:sonarqube-gradle-plugin:2.8")
-  implementation("org.jlleitschuh.gradle:ktlint-gradle:9.2.1")
-  implementation("io.gitlab.arturbosch.detekt:detekt-gradle-plugin:1.6.0")
+  implementation("org.sonarsource.scanner.gradle:sonarqube-gradle-plugin:3.0")
```
Comment (celcius112) [inline build.gradle.kts:25]: We've had some breaking changes in the ordering of a few tasks with this change. I think @fpavageau was looking into it

Comment (clemstoquart) [reply build.gradle.kts:25]: The plugin isn't using it directly. But you're right the consumers should care.

Comment (clemstoquart) [reply build.gradle.kts:25]: @celcius112 Next release will be 2.0 so it's ok if we have breaking changes.

Comment (celcius112) [reply build.gradle.kts:25]: thanks

**Pull request merged**